### PR TITLE
Handle endpoints errors

### PR DESCRIPTION
**Pull request by staltz**
@cryptix and @arj03, I already tested in production that this fixes the crashes, but it would be good to get more eyeballs looking at this, either to share context/learnings, or to check for mistakes.

### CODE REVIEW

File: src/room-observer.ts [hunk 1]
```diff
@@ -43,40 +65,67 @@ export default class RoomObserver {
     debug('announcing to portal: %s', this.roomKey);
     pull(
       this.rpc.tunnel.endpoints(),
-      (this.endpointsDrain = pull.drain((endpoints: Array<FeedId>) => {
-        const room = this.roomKey;
-        debug('got endpoints from %s: %s', room, JSON.stringify(endpoints));
+      (this.endpointsDrain = pull.drain(
+        this.endpointsUpdated,
```
Comment (cryptix) [inline src/room-observer.ts:69]: +1 for naming these

**Review by cryptix — APPROVED** [review]
can only superficially gauge this but looks better then before.

I will pull this into the tests of go-ssb-room after my lunch break.

**Comment by staltz** [comment]
I'll make a new version!

File: src/room-observer.ts [hunk 2]
```diff
@@ -43,40 +65,67 @@ export default class RoomObserver {
...
+      this.ssb.conn.stage(address, {
+        type: 'room-endpoint',
+        key,
+        room,
+        roomName: this.roomMetadata?.name,
+      });
+    }
+  };
+
+  private endpointsEnded = (err?: Error | true) => {
```
Comment (staltz) [inline src/room-observer.ts:108]: @cryptix I mostly copied this from [ssb-replicate](https://github.com/ssbc/ssb-replicate/blob/7567b077ade9350e287ae8d3a08a84ae84d8498b/legacy.js#L272)

**Comment by staltz** [comment]
@cryptix Actually you can already try `ssb-room-client@0.13.0-rc0`

File: src/room-observer.ts (hunk 2 above)
Comment (arj03) [reply src/room-observer.ts:108]: I'm fine that we add this here, and then later into some reusable module :)

**Review by arj03 — APPROVED** [review]

**Comment by staltz** [comment]
Released 0.13.0

**Pull request merged**